mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -636,4 +638,137 @@ private ArtifactRepository getArtifactRepository(String runId) {
     URI baseArtifactUri = URI.create(getRun(runId).getInfo().getArtifactUri());
     return artifactRepositoryFactory.getArtifactRepository(baseArtifactUri, runId);
   }
+
+
+  // ********************
+  // * Model Registry *
+  // ********************
+
+  /**
+   * Return the latest model version for each stage.
+   * The current stages are: [None, Staging, Production, Archived].
+   *
+   *    <pre>
+   *        List<ModelVersionDetailed> detailsList = getLatestVersions("model");
+   *
+   *        for (ModelVersionDetailed details : detailsList) {
+   *            System.out.println("Model Name: " + details.getModelVersion()
+   *                                                       .getRegisteredModel()
+   *                                                       .getName());
+   *            System.out.println("Model Version: " + details.getModelVersion().getVersion());
+   *            System.out.println("Current Stage: " + details.getCurrentStage());
+   *        }
+   *    </pre>
+   *
+   * @param modelName The name of the model
+   * @return A collection of {@link org.mlflow.api.proto.ModelRegistry.ModelVersionDetailed}
+   */
+  public List<ModelVersionDetailed> getLatestVersions(@Nonnull String modelName) {
+    String json = sendPost("registered-models/get-latest-versions",
+            mapper.makeGetLatestVersion(modelName));
+    GetLatestVersions.Response response =  mapper.toGetLatestVersionsResponse(json);
+    return response.getModelVersionsDetailedList();
+  }
+
+  /**
+   * Return the latest model version for each stage..
+   * The current stages are: [None, Staging, Production, Archived].
+   *
+   *    <pre>
+   *        ModelVersionDetailed details = getLatestVersions("model", "Staging");
+   *
+   *        System.out.println("Model Name: " + details.getModelVersion()
+   *                                                   .getRegisteredModel()
+   *                                                   .getName());
+   *        System.out.println("Model Version: " + details.getModelVersion().getVersion());
+   *        System.out.println("Current Stage: " + details.getCurrentStage());
+   *    </pre>
+   *
+   * @param modelName The name of the model
+   * @param stage The name of the stage
+   * @return The latest model version
+   *         {@link org.mlflow.api.proto.ModelRegistry.ModelVersionDetailed}
+   */
+  public ModelVersionDetailed getLatestVersions(@Nonnull String modelName, @Nonnull String stage) {
+    String json = sendPost("registered-models/get-latest-versions",
+            mapper.makeGetLatestVersion(modelName, stage));
+
+    GetLatestVersions.Response response =  mapper.toGetLatestVersionsResponse(json);
+
+    if (response.getModelVersionsDetailedCount() < 1) {
+      throw new MlflowClientException(String.format("The model %s does not exist", modelName));
```

Comment:
in which case we may not want to throw here

Reply:
Agreed. I've removed these two checks.